tests/src/tests/command_test_utils.go

```diff
@@ -0,0 +1,256 @@
+package tests
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/google/go-cmp/cmp"
+	"sigs.k8s.io/yaml"
+
+	schema "github.com/devfile/api/pkg/apis/workspaces/v1alpha2"
+)
+
+func AddEnv(numEnv int) []schema.EnvVar {
+	commandEnvs := make([]schema.EnvVar, numEnv)
+	for i := 0; i < numEnv; i++ {
+		commandEnvs[i].Name = "Name_" + GetRandomString(5, false)
+		commandEnvs[i].Value = "Value_" + GetRandomString(5, false)
+		LogMessage(fmt.Sprintf("   ....... Add Env: %s", commandEnvs[i]))
+	}
+	return commandEnvs
+}
+
+func AddAttributes(numAtrributes int) map[string]string {
+	attributes := make(map[string]string)
+	for i := 0; i < numAtrributes; i++ {
+		AttributeName := "Name_" + GetRandomString(6, false)
+		attributes[AttributeName] = "Value_" + GetRandomString(6, false)
+		LogMessage(fmt.Sprintf("   ....... Add attribute : %s = %s", AttributeName, attributes[AttributeName]))
+	}
+	return attributes
+}
+
+func addGroup() *schema.CommandGroup {
+
+	commandGroup := schema.CommandGroup{}
+
+	commandGroup.Kind = GetRandomGroupKind()
+	LogMessage(fmt.Sprintf("   ....... group Kind: %s", commandGroup.Kind))
+	commandGroup.IsDefault = GetBinaryDecision()
+	LogMessage(fmt.Sprintf("   ....... group isDefault: %t", commandGroup.IsDefault))
+
+	return &commandGroup
+}
+
+func (devfile *TestDevfile) addCommand(commandType schema.CommandType) string {
+
+	var commands []schema.Command
+	index := 0
+	if devfile.SchemaDevFile.Commands != nil {
+		// Commands already exist so expand the current slice
+		currentSize := len(devfile.SchemaDevFile.Commands)
+		commands = make([]schema.Command, currentSize+1)
+		for _, command := range devfile.SchemaDevFile.Commands {
+			commands[index] = command
+			index++
+		}
+	} else {
+		commands = make([]schema.Command, 1)
+	}
+
+	generateCommand(&commands[index], commandType)
+	devfile.SchemaDevFile.Commands = commands
+
+	return commands[index].Id
+
+}
+
+func generateCommand(command *schema.Command, commandType schema.CommandType) {
+	command.Id = GetRandomUniqueString(8, true)
+	LogMessage(fmt.Sprintf("   ....... id: %s", command.Id))
+
+	if commandType == schema.ExecCommandType {
+		command.Exec = createExecCommand()
+	} else if commandType == schema.CompositeCommandType {
+		command.Composite = createCompositeCommand()
+	}
+}
+
+func (devfile *TestDevfile) UpdateCommand(parserCommand *schema.Command) error {
+
+	errorString := ""
+	testCommand, found := getSchemaCommand(devfile.SchemaDevFile.Commands, parserCommand.Id)
+	if found {
+		LogMessage(fmt.Sprintf(" ....... Updating command id: %s", parserCommand.Id))
+		if testCommand.Exec != nil {
+			setExecCommandValues(parserCommand.Exec)
+		} else if testCommand.Composite != nil {
+			setCompositeCommandValues(parserCommand.Composite)
+		}
+		devfile.replaceSchemaCommand(*parserCommand)
+	} else {
+		errorString += LogMessage(fmt.Sprintf(" ....... Command not found in test : %s", parserCommand.Id))
```

Comment:
I would prefer to have `errorString` with type `[]string`
and do `errorString= append(errorString, errorMessage)` instead of using `+=`

Reply:
Sure append is new to me :-), but another I would like to keep for part 2.

Reply:
done